sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs

```diff
@@ -87,6 +99,116 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
             {
                 ProcessNext(message, pipeline);
             }
+
+            static string? WaitSynchronously(ValueTask<string?> headerValueTask, CancellationToken cancellationToken)
+            {
+                Task<string?> task = headerValueTask.AsTask();
+                try
+                {
+                    task.Wait(cancellationToken);
+                }
+                catch (AggregateException) { } // ignore exception here to rethrow it with EnsureCompleted
+                return task.EnsureCompleted();
```

Comment:
EnsureCompleted would wait for task to finish even if `cancellationToken` have fired, right?

Reply:
No, we won't reach this method cause `task.Wait(cancellationToken);` will throw `OperationCanceledException`, not `AggregateException`